src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -41,12 +61,8 @@ def compress(self):
         The model will be instrumented and user should never edit it after calling this method.
         `self.modules_to_compress` records all the to-be-compressed layers
         """
-        for name, module in self.bound_model.named_modules():
-            layer = LayerInfo(name, module)
-            config = self.select_config(layer)
-            if config is not None:
-                self._instrument_layer(layer, config)
-                self.modules_to_compress.append((layer, config))
+        for layer, config in self.modules_to_compress:
```

Comment:
-> 
self.detect_modules_to_compress()
for layer, config in self.modules_to_compress:

Reply:
done

Reply:
please change `self.modules_to_compress` to `modules_to_compress`

Reply:
done